istioctl/cmd/validation.go

```diff
@@ -0,0 +1,283 @@
+				time.Duration(opts.maxTimeForCheckingWebhookServer)*time.Second)
+			if err != nil {
+				fmt.Printf("err when checking webhook server: %v\n", err)
+				return err
+			}
+			webhookConfig, err := buildValidatingWebhookConfig(
+				caCert,
+				opts.webhookConfigPath,
+				opts.validatingWebhookConfigName,
+			)
+			if err != nil {
+				fmt.Printf("err when build validatingwebhookconfiguration: %v\n", err)
+				return err
+			}
+			whRet, err := createValidatingWebhookConfig(client, webhookConfig)
+			if err != nil {
+				fmt.Printf("error when creating validatingwebhookconfiguration: %v\n", err)
+				return err
+			}
+			if whRet == nil {
+				fmt.Println("validatingwebhookconfiguration created is nil")
+				return fmt.Errorf("validatingwebhookconfiguration created is nil")
+			}
+			return nil
+		},
+	}
+	validationCmd.AddCommand(cmd)
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.readCaCertFromLocalFile, "read-ca-cert-from-local-file", false, "Specifies whether reading"+
+		"CA certificate from local file system or not.")
+	// Specifies the local file path to webhook CA certificate.
+	// The default value is configured based on https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/:
+	// /var/run/secrets/kubernetes.io/serviceaccount/ca.crt.
+	flags.StringVar(&opts.caCertPath, "ca-cert-path", "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+		"Specifies the local file path to webhook CA certificate.")
+	flags.StringVar(&opts.webhookConfigPath, "config-path", "/etc/galley/validatingwebhookconfiguration.yaml",
+		"Specifies the file path of the webhook configuration.")
+	flags.StringVar(&opts.validatingWebhookConfigName, "config-name", "istio-galley",
+		"The name of the ValidatingWebhookConfiguration to manage.")
+	flags.StringVar(&opts.validatingWebhookServiceName, "service", "istio-galley",
+		"The service name of the validating webhook to manage.")
+	flags.IntVar(&opts.maxTimeForCheckingWebhookServer, "timeout", 60,
+		"	Max time (in seconds) for checking the validating webhook server. If the validating webhook server is not ready"+
+			"in the given time, exit. Otherwise, apply the webhook configuration.")
+
+	return validationCmd
+}
+
+// Create the validatingwebhookconfiguration
+func createValidatingWebhookConfig(k8sClient kubernetes.Interface,
+	config *v1beta1.ValidatingWebhookConfiguration) (*v1beta1.ValidatingWebhookConfiguration, error) {
+	if config == nil {
+		return nil, fmt.Errorf("validatingwebhookconfiguration is nil")
+	}
+	client := k8sClient.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations()
+	_, err := client.Get(config.Name, metav1.GetOptions{})
+	if err == nil {
+		// Delete the existing webhook configuration
+		if delErr := client.Delete(config.Name, &metav1.DeleteOptions{}); delErr != nil {
+			fmt.Printf("when deleting webhook configuration %v, return: %v\n", config.Name, delErr)
+		}
+		time.Sleep(5 * time.Second)
```

Comment:
What are you waiting for here? This seems brittle. Can you try to replace this with a polling loop with a timeout?

Reply:
Done: the client.Delete() and time.Sleep() has been removed. This part has been replaced with a new mechanism.